### PR TITLE
goreleaser: init at 0.145.0

### PR DESCRIPTION
**Pull request by sarcasticadmin**
###### Motivation for this change

Currently leveraging `goreleaser` for releasing go tools.

Pigging backing initial work from https://github.com/NixOS/nixpkgs/pull/82986 but it looks like its outdated. @c0deaddict FYI

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by IvarWithoutBones — APPROVED** [review]
Binary works, diff LGTM :+1: 

Result of `nixpkgs-review pr 101049` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>goreleaser</li>
  </ul>
</details>

**Review by c0deaddict — APPROVED** [review]
Nice, looks good to me! Closed my old PR.

File: pkgs/tools/misc/goreleaser/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  buildFlagsArray = [
+    "-ldflags="
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.builtBy=nixpkgs"
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/goreleaser/default.nix:24]:
```suggestion
  # tests expect the source files to be a build repo
  doCheck = false;
```

Or something similar.

File: pkgs/tools/misc/goreleaser/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = with lib; {
+    description = "Deliver Go binaries as fast and easily as possible";
+    homepage = "https://goreleaser.com";
+    maintainers = with maintainers; [ c0deaddict sarcasticadmin ];
```
Comment (endocrimes) [inline pkgs/tools/misc/goreleaser/default.nix:29]: Happy to also be a maintainer of this - I use it too, but didn't think other nix folks would be.

**Review by endocrimes — APPROVED** [review]

File: pkgs/tools/misc/goreleaser/default.nix (hunk 1 above)
Comment (sarcasticadmin) [reply pkgs/tools/misc/goreleaser/default.nix:25]: @SuperSandro2000 ✅

File: pkgs/tools/misc/goreleaser/default.nix (hunk 2 above)
Comment (sarcasticadmin) [reply pkgs/tools/misc/goreleaser/default.nix]: @endocrimes added you 🙂

**Comment by sarcasticadmin** [comment]
Thanks everyone for the approvals and feedback! I went ahead and made the appropriate updates. If theres anything else just let me know.

**Review by endocrimes — APPROVED** [review]

**Review by c0deaddict — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101049` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>goreleaser</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101049` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>goreleaser</li>
  </ul>
</details>

**Pull request merged**